x/auth/tx/query.go

```diff
@@ -93,6 +93,32 @@ func QueryTx(clientCtx client.Context, hashHexStr string) (*sdk.TxResponse, erro
 	return out, nil
 }
 
+func QueryMemTxs(clientCtx client.Context, limit int) (*sdk.UnconfirmedTxsResult, error) {
```

Comment:
there shouldnt be a need to have these items located in the auth module, they are specific to a mempool implementation.